mods/classic-taskbar-context-menu.wh.cpp

```diff
@@ -0,0 +1,591 @@
+		SetMenuItemBitmaps((HMENU)wParam, 0x1a4, MF_BYCOMMAND, MyIcons_Get(MYICON_TASKMGR), MyIcons_Get(MYICON_TASKMGR));
+		SetMenuItemBitmaps((HMENU)wParam, 0x197, MF_BYCOMMAND, MyIcons_Get(MYICON_SHOWDESKTOP), MyIcons_Get(MYICON_SHOWDESKTOP));
+
+		return ret;
+	}
+
+	return fpWndProcOld(hwnd, uMsg, wParam, lParam);
+}
+
+
+//DWORD WINAPI ThreadProc(LPVOID lpParameter);
+
+typedef BOOL (WINAPI *FPT_TrackPopupMenu)(HMENU, UINT, int, int, int, HWND, CONST RECT*);
+typedef BOOL(WINAPI *FPT_TrackPopupMenuEx)(HMENU, UINT, int, int, HWND, LPTPMPARAMS);
+
+// Pointer for calling original MessageBoxW.
+FPT_TrackPopupMenu pOriginalTrackPopupMenu;
+FPT_TrackPopupMenuEx pOriginalTrackPopupMenuEx;
+
+// Detour function which overrides TrackPopupMenu.
+BOOL WINAPI TrackPopupMenu_Hook(HMENU hMenu, UINT uFlags, int x, int y, int nReserved, HWND hWnd, CONST RECT* prcRect) {
+	if (IsWindow(hWnd)) ClassicMenuIfPossible(hWnd, hMenu);
+
+	return pOriginalTrackPopupMenu(hMenu, uFlags, x, y, nReserved, hWnd, prcRect);
+}
+
+BOOL WINAPI TrackPopupMenuEx_Hook(HMENU hMenu, UINT uFlags, int x, int y, HWND hWnd, LPTPMPARAMS lptpm) {
+	if (IsWindow(hWnd)) ClassicMenuIfPossible(hWnd, hMenu);
+	
+	return pOriginalTrackPopupMenuEx(hMenu, uFlags, x, y, hWnd, lptpm);
+}
+
+BOOL CALLBACK EnumWindowsCallBack(HWND hwnd, LPARAM lParam) {
+	if (GetWindowThreadProcessId(hwnd, NULL) == (DWORD)lParam) {
+		// Window belong to the same thread
+		char className[255];
+		GetClassNameA(hwnd, className, sizeof(className));
+
+		if (className[0] == 'A' && className[1] == 'T' && className[2] == 'L') {		
+            if (GetWindowTextLengthA(hwnd) == 0) hWnd_ATL1 = hwnd;
+            else hWnd_ATL2 = hwnd;
+		}
+	}
+
+	return TRUE;
+}
+
+
+
+
+
+
+
+
+
+
+void LoadSettings() {
+    settings.showIcons = Wh_GetIntSetting(L"showIcons");
+}
+
+
+BOOL Wh_ModInit() {
+    Wh_Log(L"Init");
+
+    LoadSettings();
+
+
+    Wh_SetFunctionHook((void*)TrackPopupMenu, (void*)TrackPopupMenu_Hook, (void**)&pOriginalTrackPopupMenu);
+    Wh_SetFunctionHook((void*)TrackPopupMenuEx, (void*)TrackPopupMenuEx_Hook, (void**)&pOriginalTrackPopupMenuEx);
+    
+
+
+	MyIcons_Load();
+
+	hWnd_TaskBar = FindWindow(TEXT("Shell_TrayWnd"), NULL);
+	// hWnd_TaskBar = *(HWND*)param;
+	if (IsWindow(hWnd_TaskBar)) {
+		OldWndProc_TaskBar = GetWindowLongPtr(hWnd_TaskBar, GWLP_WNDPROC);
+
+		if (OldWndProc_TaskBar != NULL) SetWindowLongPtr(hWnd_TaskBar, GWLP_WNDPROC, (LONG_PTR)&WndProc_TaskBar);
+	}
+
+	// The taskbar on the second screen
+	hWnd_TaskBar_SecondScreen = FindWindowA("Shell_SecondaryTrayWnd", NULL);
+	if (IsWindow(hWnd_TaskBar_SecondScreen)) {
+		OldWndProc_TaskBar_SecondScreen = GetWindowLongPtr(hWnd_TaskBar_SecondScreen, GWLP_WNDPROC);
+		
+        if (OldWndProc_TaskBar_SecondScreen != NULL) SetWindowLongPtr(hWnd_TaskBar_SecondScreen, GWLP_WNDPROC, (LONG_PTR)&WndProc_TaskBar_SecondScreen);
+	}
+
+	menu98Module = LoadLibraryA("menu98.dll");
```

Comment:
These references can be seemingly removed as well.

Reply:
I'm not sure what you mean. If you are talking about hWnd_TaskBar, it is used in the wndproc function

Reply:
I'm talking about trying to load menu98.dll which will always fail.

Reply:
Oh. why is that?

Reply:
I assumed it was part of Taskbar-Context-Menu-Tweaker, but it's seems to be a different tool, so yeah, If the user has it installed it will probably work.

Reply:
You're correct. it is not necessarily a part of taskbar context menu tweaker, but its from the same author https://github.com/rikka0w0/Menu98